graph/src/blockchain/types.rs

```diff
@@ -330,6 +341,174 @@ impl From<BlockPtr> for BlockNumber {
     }
 }
 
+fn deserialize_block_number<'de, D>(deserializer: D) -> Result<BlockNumber, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    let s: String = Deserialize::deserialize(deserializer)?;
+
+    if s.starts_with("0x") {
+        let s = s.trim_start_matches("0x");
+        i32::from_str_radix(s, 16).map_err(serde::de::Error::custom)
+    } else {
+        i32::from_str(&s).map_err(serde::de::Error::custom)
+    }
+}
+
+#[derive(Clone, PartialEq, Eq, Hash, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct BlockPtrExt {
```

Comment:
I don't know if this is a nitpick but I think we should prefer more explicit names, in this case ExternalBlockPtr if that's what it means. Would also be great to have a comments on why and how it's used because this terminology is very overloaded in some parts of the code.

Reply:
Agreed! i meant ExtendedBlockPtr, i'll go with the full name

Reply:
Resolved in latest commit